src/Http/Routing/src/Matching/JumpTableBuilder.cs

```diff
@@ -87,7 +87,9 @@ public static JumpTable Build(int defaultDestination, int exitDestination, (stri
         // Use the ILEmitTrieJumpTable if the IL is going to be compiled (not interpreted)
         if (RuntimeFeature.IsDynamicCodeCompiled)
         {
+#pragma warning disable IL3050 // See https://github.com/dotnet/linker/issues/2715.
             return new ILEmitTrieJumpTable(defaultDestination, exitDestination, pathEntries, vectorize: null, fallback);
```

Comment:
Hmm. AOT analysis recognizes pragma disable, but I don't believe it flows through to the AOT compiler.

Instead, create a local method, move `ILEmitTrieJumpTable` usage to it, and place the attribute on the local method.

Reply:
This is actually OK, because it is working around a bug in the AOT analyzer.

When the AOT compiler runs, the code inside the `if` will be trimmed away (because `IsDynamicCodeCompiled` will never be true). So the AOT compiler won't warn, since the code is trimmed.